### PR TITLE
V3: LLDP: Add packet validation (system name and description)

### PR DESCRIPTION
**Pull request by ssahani**
LLDP type system name and system description should
be with in 255 characters and unique.

Let's add the validation to discard corrupt packets.

### CODE REVIEW

**Comment by ssahani** [comment]
@keszybz Updated .

File: src/libsystemd-network/sd-lldp.c [hunk 1]
```diff
@@ -300,11 +286,45 @@ int lldp_handle_packet(tlv_packet *tlv, uint16_t length) {
                         ttl = true;
 
                         break;
+                case LLDP_TYPE_SYSTEM_NAME:
+
+                        /* According to RFC 1035 the length of a FQDN is limited to 255 characters */
+                        if (len > 255) {
+                                log_lldp("Received invalid systemd name length: %d. Dropping.", len);
```
Comment (keszybz) [inline src/libsystemd-network/sd-lldp.c:293]: Not systemD :)

Comment (keszybz) [inline src/libsystemd-network/sd-lldp.c:293]: Fixed up in merge.

Comment (ssahani) [inline src/libsystemd-network/sd-lldp.c:293]: Thanks :)

**Pull request merged**